integration_test.go

```diff
@@ -368,31 +372,273 @@ func (s *simpleService) Baz(c context.Context, in *simple.Foo) (*simple.Foo, err
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `{}`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+					Peers:       []string{"grpc://" + addr.String()},
+				},
+			},
+			wantRes: ``,
+			wantErr: "Failed while receiving stream response: code:unknown message:test error\n",
+
+			errOnStart: errors.New("test error"),
+		},
+		{
+			desc: "bidi streaming with EOF",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/BidiStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `{}`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+					Peers:       []string{"grpc://" + addr.String()},
+				},
+			},
+			wantRes:     ``,
+			wantErr:     "",
+			exitOnStart: true,
+		},
+		{
+			desc: "error on stream call due to timeout",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/BidiStream",
+					Timeout:           timeMillisFlag(time.Nanosecond),
+					RequestJSON:       `{}`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+					Peers:       []string{"grpc://" + addr.String()},
+				},
+			},
+			wantRes: ``,
+			wantErr: "Failed while making stream call",
+		},
+		{
+			desc: "server streaming",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/ServerStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `{"test":2}`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+					Peers:       []string{"grpc://" + addr.String()},
+				},
+			},
+			wantRes: `{
+  "test": 1
+}
+
+{
+  "test": 2
+}
+
+`,
+		},
+		{
+			desc: "bidirectional streaming",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/BidiStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `{"test":250}{"test":1}`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+					Peers:       []string{"grpc://" + addr.String()},
+				},
+			},
+			wantRes: `{
+  "test": 350
+}
+
+{
+  "test": 101
+}
+
+`,
+		},
+		{
+			desc: "client streaming with invalid input",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/ClientStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `{`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+					Peers:       []string{"grpc://" + addr.String()},
+				},
+			},
+			wantRes: ``,
+			wantErr: "Failed while reading stream input: unexpected EOF\n",
+		},
+	}
```

Comment:
Few missing unittests:
- Can we have unittests with yaml format?
- Can we add a test with bidi where the first values return is in a good json but not the second one? (like `{"test":250}{"test_wi": [1]}`